pallets/nodle-staking/src/lib.rs

```diff
@@ -0,0 +1,2311 @@
+                owner,
+                amount: B::default(),
+            }
+        }
+    }
+
+    impl<AccountId: Ord, Balance> Eq for Bond<AccountId, Balance> {}
+
+    impl<AccountId: Ord, Balance> PartialEq for Bond<AccountId, Balance> {
+        fn eq(&self, other: &Self) -> bool {
+            self.owner == other.owner
+        }
+    }
+
+    impl<AccountId: Ord, Balance> Ord for Bond<AccountId, Balance> {
+        fn cmp(&self, other: &Self) -> Ordering {
+            self.owner.cmp(&other.owner)
+        }
+    }
+
+    impl<AccountId: Ord, Balance> PartialOrd for Bond<AccountId, Balance> {
+        fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+            Some(self.cmp(other))
+        }
+    }
+
+    /// Just a Balance/BlockNumber tuple to encode when a chunk of funds will be unlocked.
+    #[derive(PartialEq, Eq, Clone, Encode, Decode, RuntimeDebug, Copy)]
+    pub struct UnlockChunk<Balance> {
+        /// Amount of funds to be unlocked.
+        value: Balance,
+        /// Session number at which point it'll be unlocked.
+        session_idx: SessionIndex,
+    }
+
+    #[derive(Copy, Clone, PartialEq, Eq, Encode, Decode, RuntimeDebug)]
+    /// The activity status of the validator
+    pub enum ValidatorStatus {
+        /// Committed to be online and producing valid blocks
+        Active,
+        /// Temporarily inactive
+        Idle,
+        /// Bonded until the inner round
+        Leaving(SessionIndex),
+    }
+
+    impl Default for ValidatorStatus {
+        fn default() -> ValidatorStatus {
+            ValidatorStatus::Active
+        }
+    }
+
+    #[derive(Encode, Decode, Clone, RuntimeDebug)]
+    /// Global validator state with commission fee, bonded stake, and nominations
+    pub struct Validator<AccountId, Balance> {
+        pub id: AccountId,
+        pub bond: Balance,
+        pub nomi_bond_total: Balance,
+        pub nominators: OrderedSet<Bond<AccountId, Balance>>,
+        pub total: Balance,
+        pub state: ValidatorStatus,
+        pub unlocking: Vec<UnlockChunk<Balance>>,
+    }
+
+    impl<
+            A: Ord + Clone,
+            B: AtLeast32BitUnsigned + Ord + Copy + sp_std::ops::AddAssign + sp_std::ops::SubAssign,
+        > Validator<A, B>
+    {
+        pub fn new(id: A, bond: B) -> Self {
+            let total = bond;
+            Validator {
+                id,
+                bond,
+                nomi_bond_total: Zero::zero(),
+                nominators: OrderedSet::new(),
+                total,
+                state: ValidatorStatus::default(), // default active
+                unlocking: Vec::new(),
+            }
+        }
+        pub fn is_active(&self) -> bool {
+            self.state == ValidatorStatus::Active
+        }
+        pub fn is_leaving(&self) -> bool {
+            matches!(self.state, ValidatorStatus::Leaving(_))
+        }
+        pub fn bond_more(&mut self, more: B) {
+            self.bond += more;
+            self.total += more;
```

Comment:
Let's use saturating or checked math to avoid overflows

Reply:
:+1:  Moved to saturating_add()